src/testdir/test_delete.vim

```diff
@@ -97,3 +97,48 @@ func Test_symlink_recursive_delete()
   call assert_equal(0, delete('Xdir4/Xfile'))
   call assert_equal(0, delete('Xdir4', 'd'))
 endfunc
+
+func Test_complicated_name_recursive_delete()
+  call mkdir('Xcomplicated')
+  call mkdir('Xcomplicated/[complicated-1 ]')
+  call mkdir('Xcomplicated/{complicated,2 }')
+  split Xcomplicated/Xfile
+  call setline(1, ['a', 'b'])
+  w
+  w Xcomplicated/\[complicated-1\ \]/Xfile
+  w Xcomplicated/\{complicated,2\ \}/Xfile
```

Comment:
As far as I understand, E16 is caused by expanding the filename and interpreting the square brackets as part of a regular expression. That's what my patch is supposed to solve for delete()... 

Maybe the easiest solution is to make the whole test unix-only... Opinions on this would be welcome.